File: week7/coalery/94.cpp
// Time Complexity : O(N)
// Traversal All Node, So O(N)

class Solution {
private:
    vector<int> v;
    void _inorder(TreeNode* cursor) {
        if(cursor == nullptr) return;
        _inorder(cursor->left);
        v.push_back(cursor->val);
        _inorder(cursor->right);
    }
public:
    vector<int> inorderTraversal(TreeNode* root) {
        _inorder(root);
        return v;
    }
};

File: week7/nibble/94.js
var inorderTraversal = function (root) {
    let answer = [];

    function inorder(root) {
        if (!root) return;
        inorder(root.left);
        answer.push(root.val);
        inorder(root.right);
    }

    inorder(root);
    return answer;
};